migrations/20250227_viz_move/migration.py

```diff
@@ -0,0 +1,249 @@
+    logger.info("Renaming microreact folders and files to visualise...")
+    updated_folders = all_folders.copy()
+
+    for i, folder_path in enumerate(all_folders):
+        folder_name = os.path.basename(folder_path)
+        if folder_name.startswith("microreact_"):
+            new_folder_name = folder_name.replace("microreact_", "visualise_")
+            new_folder_path = os.path.join(
+                os.path.dirname(folder_path), new_folder_name
+            )
+            if not os.path.exists(new_folder_path):
+                os.rename(folder_path, new_folder_path)
+                logger.info(
+                    f"Renamed folder: {folder_name} → {new_folder_name}"
+                )
+                # Update the folder path in the list
+                updated_folders[i] = new_folder_path
+
+            # Rename files within the folder
+            for root, dirs, files in os.walk(new_folder_path):
+                for file in files:
+                    if file.startswith("microreact_"):
+                        new_file_name = file.replace(
+                            "microreact_", "visualise_", 1
+                        )
+                        new_file = os.path.join(root, new_file_name)
+                        old_file = os.path.join(root, file)
+                        if not os.path.exists(new_file):
+                            os.rename(old_file, new_file)
+                            logger.info(
+                                f"Renamed file: {file} → {new_file_name}"
+                            )
+
+    return updated_folders
+
+
+def move_network_files_to_visualise(all_folders, logger):
+    """Move network files to visualise folders."""
+    logger.info("Moving network files to visualise folders...")
+    for folder_path in all_folders:
+        folder_name = os.path.basename(folder_path)
+        if folder_name == "network":
+            parent_folder = os.path.dirname(folder_path)
+
+            # Process CSV files
+            move_csv_files(folder_path, parent_folder, logger)
+
+            # Process GraphML files
+            move_graphml_files(folder_path, parent_folder, logger)
+
+
+def move_csv_files(network_folder, parent_folder, logger):
+    """Move CSV files from network folder to visualise folders."""
+    for file in os.listdir(network_folder):
+        if file.endswith(".csv"):
+            for dir_name in os.listdir(parent_folder):
+                if os.path.isdir(
+                    os.path.join(parent_folder, dir_name)
+                ) and dir_name.startswith("visualise_"):
+                    # Rename file
+                    new_fname = file.replace("network", dir_name)
+                    new_fname_path = os.path.join(network_folder, new_fname)
+
+                    if not os.path.exists(new_fname_path):
+                        shutil.copyfile(
+                            os.path.join(network_folder, file), new_fname_path
```

Comment:
So this is renaming it in the network folder, and then it subsequently it moves it to the visualise folder? Why not just do both of those things in one copyfile?

Reply:
yeah i realised that later im stupid... il just leave as its working and I don't wanna break it... (as we don't have to maintain this code shouldn't be a biggie)